src/plugins/loopVideos/index.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2023 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { definePluginSettings } from "@api/Settings";
+import ErrorBoundary from "@components/ErrorBoundary";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { Tooltip, useEffect, useRef, useState } from "@webpack/common";
+
+
+const settings = definePluginSettings({
+    loop: {
+        description: "Whether to enable looping by default or not",
+        type: OptionType.BOOLEAN,
+        default: true
+    }
+});
+
+export default definePlugin({
+    name: "LoopVideos",
+    description: "Adds a loop button next to the volume slider on embedded videos",
+    authors: [Devs.Lumap],
+    settings,
+
+    patches: [
+        {
+            find: ".videoControls:",
+            replacement: {
+                match: /children:\[this\.renderPlayIcon\(\),.{0,200}\.setDurationRef}\),/,
```

Comment:
why match so much stuff u don't need?
```suggestion
                match: /ref:this\.setDurationRef}\),/,
```